test/mocha/30-backwards-compatibility.js

```diff
@@ -0,0 +1,536 @@
+      const handle = 'test';
+      capabilityAgent = await CapabilityAgent.fromSecret({secret, handle});
+
+      // create keystore for capability agent
+      const keystoreAgent = await helpers.createKeystoreAgent(
+        {capabilityAgent});
+      // generate key for signing VCs (make it a did:key DID for simplicity)
+      const assertionMethodKey = await keystoreAgent.generateKey({
+        type: 'asymmetric',
+        publicAliasTemplate: 'did:key:{publicKeyMultibase}#' +
+              '{publicKeyMultibase}'
+      });
+
+      // create EDV for storage (creating hmac and kak in the process)
+      const {
+        edvConfig,
+        hmac,
+        keyAgreementKey
+      } = await helpers.createEdv({capabilityAgent, keystoreAgent});
+
+      // get service agent to delegate to
+      const serviceAgentUrl =
+          `${baseUrl}/service-agents/${encodeURIComponent(serviceType)}`;
+      const {data: serviceAgent} = await httpClient.get(
+        serviceAgentUrl, {agent});
+
+      // delegate edv, hmac, and key agreement key zcaps to service agent
+      const {id: edvId} = edvConfig;
+      zcaps.edv = await helpers.delegate({
+        controller: serviceAgent.id,
+        delegator: capabilityAgent,
+        invocationTarget: edvId
+      });
+      const {keystoreId} = keystoreAgent;
+      zcaps.hmac = await helpers.delegate({
+        capability: `urn:zcap:root:${encodeURIComponent(keystoreId)}`,
+        controller: serviceAgent.id,
+        invocationTarget: hmac.id,
+        delegator: capabilityAgent
+      });
+      zcaps.keyAgreementKey = await helpers.delegate({
+        capability: `urn:zcap:root:${encodeURIComponent(keystoreId)}`,
+        controller: serviceAgent.id,
+        invocationTarget: keyAgreementKey.kmsId,
+        delegator: capabilityAgent
+      });
+      zcaps[`assertionMethod:${assertionMethodKey.algorithm}`] = await helpers
+        .delegate({
+          capability: `urn:zcap:root:${encodeURIComponent(keystoreId)}`,
+          controller: serviceAgent.id,
+          invocationTarget: assertionMethodKey.kmsId,
+          delegator: capabilityAgent
+        });
+
+      // create issuer instance w/ no status list options
+      const noStatusListIssuerConfig = await helpers.createConfig(
+        {capabilityAgent, zcaps, suiteName: 'Ed25519Signature2020'});
+      noStatusListIssuerId = noStatusListIssuerConfig.id;
+      noStatusListIssuerRootZcap =
+          `urn:zcap:root:${encodeURIComponent(noStatusListIssuerId)}`;
+
+      // Intentionally change the referenceId of the assertion method zcap
+      // in the database to be lowercase
+      const serviceCoreConfigCollection =
+         database.collections['service-core-config-vc-issuer'];
+      await serviceCoreConfigCollection.updateOne({
+        'config.id': noStatusListIssuerId,
+      }, {
+        $rename: {
+          'config.zcaps.assertionMethod:Ed25519':
+             'config.zcaps.assertionMethod:ed25519'
+        }
```

Comment:
I see, nice. If we don't have anything that checks to make sure that when you fetch the config it actually uses the lower case thing here -- we should add that too just to make sure that this `updateOne` is doing what it's supposed to do. We could just fetch the config from the collection right after to do that check.

Reply:
Updated [414df1f](https://github.com/digitalbazaar/bedrock-vc-issuer/pull/113/commits/414df1f4d55ca14a952a2369c6ad808c9711b3a1)